From 97486a27dc3ad7b4fb4486c83755db83dee5a33e Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Fri, 17 Jun 2016 10:00:07 -0700
Subject: [PATCH 0630/1164] CLOUDERA-BUILD: CDH-41470 : patch for HIVE-4570
 removes protected fields which can break
 dependencies

Backport for HIVE-4570 changes some protected fields in the class Task.java which breaks the sentry code. This change adds them back again and sets the fields using the new implementation

Change-Id: Ib41e6f38ac488cd7f440b3dc798f9e31382b4dc9
---
 .../java/org/apache/hadoop/hive/ql/exec/Task.java  |   34 ++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/Task.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/Task.java
index 8d60fcd..9724c78 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/Task.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/Task.java
@@ -56,6 +56,14 @@
   private static final long serialVersionUID = 1L;
   public transient HashMap<String, Long> taskCounters;
   public transient TaskHandle taskHandle;
+  @Deprecated
+  protected transient boolean started;
+  @Deprecated
+  protected transient boolean initialize;
+  @Deprecated
+  protected transient boolean isdone;
+  @Deprecated
+  protected transient boolean queued;
   protected transient HiveConf conf;
   protected transient Hive db;
   protected transient LogHelper console;
@@ -123,6 +131,10 @@
   private Throwable exception;
 
   public Task() {
+    isdone = false;
+    started = false;
+    initialize = false;
+    queued = false;
     this.taskCounters = new HashMap<String, Long>();
     taskTag = Task.NO_TAG;
   }
@@ -133,6 +145,8 @@ public TaskHandle getTaskHandle() {
 
   public void initialize(HiveConf conf, QueryPlan queryPlan, DriverContext driverContext) {
     this.queryPlan = queryPlan;
+    isdone = false;
+    started = false;
     setInitialized();
     this.conf = conf;
     try {
@@ -160,6 +174,26 @@ private void updateStatusInQueryDisplay() {
 
   private void setState(TaskState state) {
     this.taskState = state;
+    // Switch block below is needed to update the state of deprecated fields
+    // to maintain backward compatibility
+    // this should be removed in newer version of hive when the deprecated
+    // fields can be safely removed
+    switch (taskState) {
+        case RUNNING:
+            this.started = true;
+            break;
+        case INITIALIZED:
+            this.initialize = true;
+            break;
+        case QUEUED:
+            this.queued = true;
+            break;
+        case FINISHED:
+            this.isdone = true;
+            break;
+        default:
+            break;
+    }
     updateStatusInQueryDisplay();
   }
   /**
-- 
1.7.9.5

